src/globals/grid/_grid.scss

```diff
@@ -0,0 +1,69 @@
+$gutter-size: 10px;
+$i: 0;
+
+@mixin column-size($col-num) {
+  flex-basis: (100% * $col-num / 12);
+  max-width: (100% * $col-num / 12);
+}
+
+.bx--grid {
+  margin: 0 3%;
+
+  @include breakpoint('bp--xs--major') {
+    margin: 0 5%;
+  }
+}
+
+.bx--row {
+  display: flex;
+  flex-wrap: wrap;
+  margin: 0 $gutter-size * -1;
+}
+
+[class*="bx--col"] {
+  @include reset;
+  padding: 0 $gutter-size;
+  flex: 1;
+  flex-basis: 100%;
+}
+
+@while $i <= 12 {
+  @media (max-width: 720px) {
+    .bx--col-xs#{$i} {
+      @include column-size($i);
+    }
+  }
+
+  @media (min-width: 720px) {
```

Comment:
two small things

- we should probably use/define the breakpoints and throw it up into our global variables for theming purposes
- with this `while` loop, you'll need to add a max-width to each of these for specificity. Otherwise `.col-xs-12` is defined after `.col-md-6`, so `.col-xs-12` takes precedence